src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/json/EventTransformer.scala

```diff
@@ -35,18 +28,82 @@ import org.json4s.JsonDSL._
 // Jackson
 import com.fasterxml.jackson.core.JsonParseException
 
+// This library
+import JsonShredder.{ Contexts, UnstructEvent }
+
+
 /**
  * Object to xxx
  */
 object EventTransformer {
 
-  private val StringField: TsvToJsonConverter = (key, value) => JObject(key -> JString(value)).successNel
-  private val IntField: TsvToJsonConverter = (key, value) => JObject(key -> JInt(value.toInt)).successNel
+  /**
+   * The event as a shredded stringified JSON along with it's inventory
+   * (set of JSON fields)
+   *
+   * @param event stringified event JSON
+   * @param inventory set of JSON fields (contexts, unsturct event)
+   */
+  case class EventWithInventory(event: String, inventory: Set[String])
+
+  /**
+   * Convert a string with Enriched event TSV to a JSON string
+   *
+   * @param line enriched event TSV line
+   * @return ValidatedRecord for the event
+   */
+  def transform(line: String): ValidatedEvent = {
+    // The -1 is necessary to prevent trailing empty strings from being discarded
+    jsonifyGoodEvent(line.split("\t", -1)).map(json => compact(json))
+  }
+
+  /**
+   * Convert an Amazon Kinesis record to a JSON string
```

Comment:
Let's update this function to return the Set of Iglu URIs, not the fixSchema'ed strings.